src/preprocessing/passes/bv_ackermann.cpp

```diff
@@ -41,11 +42,54 @@ namespace passes {
 namespace
 {
 
-void storeFunction(
-    TNode func,
-    TNode term,
-    FunctionToArgsMap& fun_to_args,
-    SubstitutionMap& fun_to_skolem)
+void addLemmaForPair(TNode args1,
+                     TNode args2,
+                     const TNode func,
+                     AssertionPipeline* assertionsToPreprocess,
+                     NodeManager* nm,
+                     std::stack<TNode>* stack)
+{
+  Node args_eq;
+
+  if (args1.getKind() == kind::APPLY_UF)
+  {
+    AlwaysAssert(args1.getKind() == kind::APPLY_UF
```

Comment:
Let's get rid of the first part of this AlwaysAssert as it is the same as the if-condition